src/Snackbar/Snackbar.react.js

```diff
@@ -0,0 +1,207 @@
+/* eslint-disable import/no-unresolved, import/extensions */
+import React, { PropTypes, PureComponent } from 'react';
+import { View, Text, Animated, Easing, Platform, StyleSheet } from 'react-native';
+
+import Button from '../Button';
+
+const propTypes = {
+    /**
+    * The text message to display.
+    */
+    message: PropTypes.string.isRequired,
+    /**
+    * Whether or not the snackbar is visible.
+    */
+    visible: PropTypes.bool.isRequired,
+    /**
+    * The amount of time in milliseconds to show the snackbar.
+    */
+    timeout: PropTypes.number.isRequired,
+    /**
+    * Callback for when the timeout finishes.
+    */
+    onRequestClose: PropTypes.func.isRequired,
+    /**
+    * Whether or not there is a bottom navigation on the screen.
+    */
+    bottomNavigation: PropTypes.bool.isRequired,
+    /**
+    * The function to execute when the action is clicked.
+    */
+    actionHandler: PropTypes.func,
+    /**
+    * The function to execute when the action is clicked.
+    */
+    actionText: PropTypes.string,
+    /**
+    * Inline style of snackbar
+    */
+    style: PropTypes.shape({
+        container: View.propTypes.style,
+        message: View.propTypes.style,
+        actionContainer: View.propTypes.style,
+        actionText: View.propTypes.style,
+    }),
+};
+const defaultProps = {
+    visible: false,
+    timeout: 2750,
+    bottomNavigation: false,
+    style: {},
+};
+const contextTypes = {
+    uiTheme: PropTypes.object.isRequired,
+};
+
+function getStyles(props, context) {
+    const { snackbar } = context.uiTheme;
+    const local = {};
+
+    return {
+        container: [
+            snackbar.container,
+            local.container,
+            props.style.container,
+        ],
+        message: [
+            snackbar.message,
+            local.message,
+            props.style.message,
+        ],
+        actionContainer: [
+            snackbar.actionContainer,
+            local.actionContainer,
+            props.style.actionContainer,
+        ],
+        actionText: [
+            snackbar.actionText,
+            local.actionText,
+            props.style.actionText,
+        ],
+    };
+}
+
+/**
+* Component for snackbars
+* https://material.io/guidelines/components/snackbars-toasts.html
+*/
+class Snackbar extends PureComponent {
+    constructor(props, context) {
+        super(props, context);
+        const styles = getStyles(props, context);
```

Comment:
I like it man :)

I've been thinking about this for couple of weeks :) And I want to make a revision for all components .. Unfortunately, you will have a problem with this, since you don't care about changes of style prop in `componentWillReceiveProps` .. 

I was thinking about some method like `shouldStyleUpdate` .. returns true if any property that can affect style was chaged .. (style, size in IconToggle, etc.)